src/hex/shape.rs

```diff
@@ -1,21 +1,43 @@
-use std::ops::AddAssign;
+use std::rc::Rc;
 
 use crate::core::transform::{Transform, Transformable};
 
 use super::coordinate::Axial;
 
+#[derive(Clone, PartialEq, Debug, Default)]
+pub struct HexTile<T> {
+    pub coord: Axial,
+    pub data: Option<Rc<T>>,
+}
+
+impl<T: Default> HexTile<T> {
+    pub fn new(coord: Axial, data: Option<T>) -> HexTile<T> {
+        HexTile { coord, data: data.map(|value| Rc::new(value))}
+    }
+}
+
+impl<T: Default> From<Axial> for HexTile<T> {
```

Comment:
Will this at some point need to have the ability to populate the data field with something other than a `None`?

Reply:
Since the field is public so long as you get a mutable reference to the tile, you can assign a value to data.